### PR TITLE
fix(signature-collection): refactor

### PR DESCRIPTION
**Pull request by alexdiljar**
- Removing current collection guard to reduce stress on api
- Updates following that change
- new bulk upload error
- collection status in object

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/signature-collection/src/lib/guards/userAccess.guard.ts [hunk 1]
```diff
@@ -40,7 +45,7 @@ export class UserAccessGuard implements CanActivate {
       return isDelegatedUser ? false : true
     }
 
-    const signee: SignatureCollectionSignee = request.body.signee
+    // const signee: SignatureCollectionSignee = request.body.signee
```
Comment (albinagu) [inline libs/api/domains/signature-collection/src/lib/guards/userAccess.guard.ts:48]: remove?

File: libs/clients/signature-collection/src/lib/signature-collection-admin.service.ts [hunk 2]
```diff
@@ -146,18 +147,18 @@ export class SignatureCollectionAdminClientService {
...
+    // if (collectionStatus === CollectionStatus.Processing) {
+    const collection = await this.getApiWithAuth(
+      this.collectionsApi,
+      auth,
+    ).medmaelasofnunIDToggleSofnunPost({
+      iD: parseInt(collectionId),
+    })
+    return { success: !!collection }
+    // }
+    // return { success: false }
```
Comment (albinagu) [inline libs/clients/signature-collection/src/lib/signature-collection-admin.service.ts:161]: comments hér

File: libs/clients/signature-collection/src/lib/types/bulkUpload.dto.ts [hunk 3]
```diff
@@ -42,5 +42,9 @@ export const mapBulkResponse = (
       nationalId,
       reason: 'Ekki með íslenskt ríkisfang',
     })) ?? []),
+    ...(signatures.ekkiBuseta?.map((nationalId) => ({
+      nationalId,
+      reason: 'Ekki með íslenskan búsetu',
```
Comment (albinagu) [inline libs/clients/signature-collection/src/lib/types/bulkUpload.dto.ts:47]: íslenska* ?

File: libs/clients/signature-collection/src/lib/signature-collection-admin.service.ts [hunk 4]
```diff
@@ -146,18 +70,15 @@ export class SignatureCollectionAdminClientService {
...
-        this.collectionsApi,
-        auth,
-      ).medmaelasofnunIDToggleSofnunPost({
-        iD: parseInt((await this.currentCollection()).id),
-      })
-      return { success: !!collection }
-    }
-    return { success: false }
+  async processCollection(collectionId: string, auth: Auth): Promise<Success> {
+    // const collectionStatus = await this.collectionStatus(auth)
```
Comment (albinagu) [inline libs/clients/signature-collection/src/lib/signature-collection-admin.service.ts:74]: comment?

File: libs/clients/signature-collection/src/lib/signature-collection-admin.service.ts [hunk 5]
```diff
@@ -176,45 +97,26 @@ export class SignatureCollectionAdminClientService {
...
   ): Promise<Slug> {
-    const { id } = await this.currentCollection()
+    const { id, areas: collectionAreas } = await this.currentCollection(auth)
     // check if collectionId is current collection and current collection is open
     if (collectionId !== id) {
       throw new Error('Collection id input wrong')
     }
 
-    const collectionAreas = await this.getAreas(id)
+    // const collectionAreas = collection.ar //await this.getAreas(id)
```
Comment (albinagu) [inline libs/clients/signature-collection/src/lib/signature-collection-admin.service.ts:116]: ?

File: libs/clients/signature-collection/src/lib/signature-collection.service.ts [hunk 6]
```diff
@@ -107,10 +106,10 @@ export class SignatureCollectionClientService {
       throw new Error('User is already owner of lists')
     }
 
-    const collectionAreas = await this.getAreas(id)
+    // const collectionAreas = await this.getAreas(id)
```
Comment (albinagu) [inline libs/clients/signature-collection/src/lib/signature-collection.service.ts:109]: //

File: libs/clients/signature-collection/src/lib/types/collection.dto.ts [hunk 7]
```diff
@@ -24,6 +24,47 @@ export interface Collection {
...
+    return CollectionStatus.InitialActive
+  }
+
+  // Initial opening time passed not all lists reviewed
+  if (!hasActiveLists && !processed) {
+    return CollectionStatus.InInitialReview
+  }
+  // Initial opening time passed all lists have been reviewd
+  // Not in use for now check prosseced status in frontend
+  // if (!hasActiveLists && !processed && !hasInReview) {
```
Comment (albinagu) [inline libs/clients/signature-collection/src/lib/types/collection.dto.ts:52]: taka út?

**Review by albinagu — APPROVED** [review]

**Review by RunarVestmann — APPROVED** [review]

**Pull request merged**